dartagnan/src/main/java/com/dat3m/dartagnan/program/analysis/alias/InclusionBasedPointerAnalysis.java

```diff
@@ -179,7 +184,7 @@ private void processWriter(RegWriter event) {
         } else if (event instanceof Load load) {
             final Offset<Variable> address = getResultVariable(load.getAddress(), load);
             if (address == null) {
-                logger.warn("null pointer address for {}", event);
+                logger.warn("null pointer address for {}: {}", event.getMetadata(SourceLocation.class), event);
```

Comment:
You can use the static method `SyntacticContextAnalysis.getSourceLocationString(Event)` to get a better location message and a proper one even if there is no source location metadata attached.